library/core/src/iter/adapters/skip.rs

```diff
@@ -43,11 +43,14 @@ where
     fn nth(&mut self, n: usize) -> Option<I::Item> {
         if self.n > 0 {
             // Checked add to handle overflow case.
-            let n = self.n.checked_add(n).unwrap_or_else(|| {
-                // In case of overflow, load skip value, then load `n`.
-                self.iter.nth(self.n);
-                n
-            });
+            let n = match self.n.checked_add(n) {
+                Some(n) => n,
+                None => {
+                    // In case of overflow, load skip value, then load `n`.
+                    self.iter.nth(self.n)?;
+                    n
+                }
+            };
             self.n = 0;
```

Comment:
You aren't setting `self.n = 0` if `self.iter.nth(self.n)` returns `None` due to the early return.
```suggestion
            let n = match self.n.checked_add(n) {
                Some(n) => n,
                None => {
                    // In case of overflow, load skip value, then load `n`.
                    self.iter.nth(crate::mem::take(&mut self.n))?;
                    n
                }
            };
```

Reply:
you're right, I shouldn't have been so quick to update